applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.hpp

```diff
@@ -178,23 +271,36 @@ namespace Kratos
 
 		IntegrationMethod GetIntegrationMethod() const override;
 
+
+		/**
+         * @brief This function calculates nodal moments due to self-weight
+		 * @param ForceInput The self-weight line load vector
+		 * @param rRightHandSideVector The right hand side of the problem
+		 * @param GeometryLength The element length
+         */		
 		void CalculateAndAddWorkEquivalentNodalForcesLineLoad(
 			const bounded_vector<double,msDimension> ForceInput,
 			bounded_vector<double,msElementSize>& rRightHandSideVector,
 			const double GeometryLength);
 
-
+		/**
+         * @brief This function calculates the geometric part of the total stiffness matrix
+         */
 		void CalculateGeometricStiffnessMatrix(MatrixType& rGeometricStiffnessMatrix,
 			ProcessInfo& rCurrentProcessInfo);
 
+
+		/**
+         * @brief This function calculates the elastic part of the total stiffness matrix
+         */
 		void CalculateElasticStiffnessMatrix(MatrixType& rElasticStiffnessMatrix,
 			ProcessInfo& rCurrentProcessInfo);
 
 	private:
 		Vector mNX, mNY, mNZ;
 		Vector mTotalNodalDeformation;

```

Comment:
what is this needed for? as a minimum comment here what it should cotain.
also ideally do not store this but recompute it as needed (if that is possible)